Fix the separator issue, support multi guest test

RoyCai7 opened the pull request:
Issue Description:
When attempting to install a guest, the LTSS (Long Term Service Support) registration code cannot be get. Instead, it displays an incorrect or empty string '-E "|||||||||||||"'.


- Related ticket: https://progress.opensuse.org/issues/169678
- Needles: N/A
- Verification run: 
       - [multi-guest test](http://openqa.qa2.suse.asia/tests/80419) with '|'
       - [multi-guest test](http://openqa.qa2.suse.asia/tests/80465) with ','
       - [one guest test](http://openqa.qa2.suse.asia/tests/80466)

Julie-CAO commented:
thanks for the fix. Previously I used "|" in ltss tests to pass multiple scc_codes to multiple guests but failed, all guests got the last scc_dode. I assume your fix will fix this?

waynechen55 commented:
@RoyCai7  Would you help also have a second look at your previous pull request https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20501 ?

I think we'd better have a second thought, decide the potential root cause and fix there. I think if reg_code_ltss is retrieved the same as ```reg_code``` in ```get_guest_code```, it should return the correct result with separator ```|```.  
And I do not think you need ```get_ltss_regcode``` function.
Let's clean all these things up here.

RoyCai7 commented:
> thanks for the fix. Previously I used "|" in ltss tests to pass multiple scc_codes to multiple guests but failed, all guests got the last scc_dode. I assume your fix will fix this?
Will make change to be suitable for you job.

RoyCai7 commented:
> @RoyCai7 Would you help also have a second look at your previous pull request #20501 ?
> 
> I think we'd better have a second thought, decide the potential root cause and fix there. I think if `regcode_ltss` is retrieved the same way as `regcode` in `get_guest_regcode` function, it should return the correct result with separator `|`.
> 
> And I do not think you need `get_ltss_regcode` function ??? Better remove it ???
> 
> Let's clean all these things up here. What I am trying to convey is we'd better do not do fix on fix . Hope you can understand.
> 
> By the way, I think the potential root cause lies here in `guest_guest_regcode`:
> 
> ```
> my @guest_parts = split($args{separator}, $guest);
> ```
> 
> should be
> 
> ```
> my @guest_parts= $args{separator} eq '|' ? split("\\$args{separator}", $guest) : split("$args{separator}", $guest);
> ```
> 
> You may forgot using escape character I think.
> 
> So I think, maybe:
> 
> * Make above suggested change ???
> * Remove ``get_ltss_regcode``` ???
> 
> are the changes need to be made in this pull request instead of current change.

Good solution, Will make improvement.

waynechen55 commented:
If you passed a test with multiple guests with separators, LGTM. Ensure you tested what you changed.

RoyCai7 commented:
> If you passed a test with multiple guests with separators, LGTM. Ensure you tested what you changed.

Passed test, I think it works well.
